cpp/src/io/avro/reader_impl.cu

```diff
@@ -407,8 +420,9 @@ table_with_metadata reader::impl::read(avro_reader_options const &options, cudaS
         for (const auto &sym : col_schema.symbols) { dictionary_data_size += sym.length(); }
       }
 
-      hostdevice_vector<uint8_t> global_dictionary(
-        total_dictionary_entries * sizeof(gpu::nvstrdesc_s) + dictionary_data_size);
+      hostdevice_vector<gpu::nvstrdesc_s> global_dictionary(total_dictionary_entries +
```

Comment:
I feel like this change would've broken the memcpy below. `global_dictionary.host_ptr() + dict_pos` has a different meaning now. I don't know why it didn't break. Insufficient tests? Can you also investigate that?

Reply:
Hmm. Looks like we have insufficient testing in this area. Maybe we should improve test coverage before moving ahead with this PR.